client/my-sites/domains/domain-management/list/helpers.ts

```diff
@@ -15,3 +36,9 @@ export const filterDomainsByOwner: FilterDomainsByOwnerType = ( domains, filter
 		return true;
 	} );
 };
+
+export const filterDomainsDomainOnly: FilterDomainsDomainOnlyType = ( domains, sites ) => {
```

Comment:
Nit: Maybe `filterDomainOnlyDomains` could be a more natural name for this function? 🙂